terraform/projects/app-ecs-instances/instance-user-data.tpl

```diff
@@ -32,10 +32,15 @@ else
   echo "[$(date '+%H:%M:%S %d-%m-%Y')] attach-volume: /dev/$DEVICE is already formatted: $(file -s /dev/$DEVICE)"
 fi
 
-# Allow prometheus container user access to read/write/execute within container
+# Allow prometheus container user access to read/write metrics to persistent volume mount from within a container
 mkdir -p /ecs/prometheus_data
+
+#Cretate prometheus & grant access to prometheus data files. We create the group & match it to the gid
+groupadd --system --gid 65534 prometheus
+useradd --uid 65534 --gid 65534 --system --groups prometheus
```

Comment:
Glancing at the docs (https://linux.die.net/man/8/useradd), you might be able to do the groupadd and useradd in one line using `useradd --usergroup`. Maybe worth a quick look (and a double check that is correct) but not fussed if you don't use it (you might also feel it impacts the readability)

Reply:
You can not set GID this way. This is why we had to do this in this way.